src/_pytest/main.py

```diff
@@ -770,3 +741,59 @@ def genitems(
                 for subnode in rep.result:
                     yield from self.genitems(subnode)
             node.ihook.pytest_collectreport(report=rep)
+
+
+def search_pypath(module_name: str) -> str:
+    """Search sys.path for the given a dotted module name, and return its file system path."""
+    try:
+        spec = importlib.util.find_spec(module_name)
+    # AttributeError: looks like package module, but actually filename
+    # ImportError: module does not exist
+    # ValueError: not a module name
+    except (AttributeError, ImportError, ValueError):
+        return module_name
+    if spec is None or spec.origin is None or spec.origin == "namespace":
+        return module_name
+    elif spec.submodule_search_locations:
+        return os.path.dirname(spec.origin)
+    else:
+        return spec.origin
+
+
+def resolve_path_argument(
```

Comment:
I think "path argument" is perhaps *slightly* misleading given that it can be a dotted python path and the selection part.

Maybe `resolve_selection_argument`? `resolve_collection_argument`?

Reply:
Good point, went with `resolve_collection_argument`. 👍